service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java

```diff
@@ -0,0 +1,213 @@
+            .build();
+    }
+
+    private String getUsername(Long userID) {
+        User currentUser = userRepository.getOne(userID);
+        return currentUser.getRecipientName() + " " + currentUser.getRecipientSurname();
+    }
+
+    @Override
+    public void addUserViolation(AddingViolationsToUserDto add, MultipartFile[] multipartFiles, String uuid) {
+        Order order = orderRepository.findById(add.getOrderID()).orElseThrow(() -> new UnexistingOrderException(
+            ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST));
+        User currentUser = userRepository.findUserByUuid(uuid)
+            .orElseThrow(() -> new UserNotFoundException(USER_WITH_CURRENT_ID_DOES_NOT_EXIST));
+        if (violationRepository.findByOrderId(order.getId()).isEmpty()) {
+            User user = order.getUser();
+            Violation violation = violationBuilder(add, order);
+            if (multipartFiles.length > 0) {
+                List<String> images = new LinkedList<>();
+                setImages(multipartFiles, images);
+                violation.setImages(images);
+            }
+            violationRepository.save(violation);
+            user.setViolations(userRepository.countTotalUsersViolations(user.getId()));
+            userRepository.save(user);
+            eventService.save(OrderHistory.ADD_VIOLATION, currentUser.getRecipientName()
+                + "  " + currentUser.getRecipientSurname(), order);
+            notificationService.notifyAddViolation(order);
+        } else {
+            throw new OrderViolationException(ORDER_ALREADY_HAS_VIOLATION);
+        }
+    }
+
+    private Violation violationBuilder(AddingViolationsToUserDto add, Order order) {
+        return Violation.builder()
+            .violationLevel(ViolationLevel.valueOf(add.getViolationLevel().toUpperCase()))
+            .description(add.getViolationDescription())
+            .violationDate(order.getOrderDate())
+            .order(order)
+            .build();
+    }
+
+    /**
+     * Method returns detailed information about user violation by order id.
+     *
+     * @param orderId of {@link Long} order id;
+     * @return {@link ViolationDetailInfoDto};
+     * @author Rusanovscaia Nadejda
+     */
+
+    @Override
+    @Transactional
+    public Optional<ViolationDetailInfoDto> getViolationDetailsByOrderId(Long orderId) {
+        User user =
+            userRepository.findUserByOrderId(orderId).orElseThrow(() -> new NotFoundException(EMPLOYEE_NOT_FOUND));
+        return violationRepository.findByOrderId(orderId).map(v -> ViolationDetailInfoDto.builder()
+            .orderId(orderId)
+            .userName(user.getRecipientName())
+            .violationLevel(v.getViolationLevel())
+            .description(v.getDescription())
+            .images(v.getImages())
+            .violationDate(v.getViolationDate())
+            .build());
+    }
+
+    @Override
+    @Transactional
+    public void deleteViolation(Long id, String uuid) {
```

Comment:
id and uuid argument names are not self descriptive. Give them better names to understand the context